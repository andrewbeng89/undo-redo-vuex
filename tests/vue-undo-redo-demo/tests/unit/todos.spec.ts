import { mount } from "@vue/test-utils";
import Todos from "../../src/components/Todos/Todos.vue";

test("renders correctly", () => {
  const context = {
    props: {
      list: [],
      label: "You don't have any Todos yet",
      canUndo: false,
      canRedo: false
    }
  };
  const listeners = {
    postNewTodo: () => {},
    undo: () => {},
    redo: () => {}
  };

  const wrapper = mount(Todos, {
    context,
    listeners
  });
  expect(wrapper.element).toMatchSnapshot();
});
