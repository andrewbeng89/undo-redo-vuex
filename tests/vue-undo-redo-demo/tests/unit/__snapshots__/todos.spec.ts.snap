// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button>
    Add
  </button>
   
  <button
    disabled="disabled"
    id="undo"
  >
    Undo
  </button>
   
  <button
    disabled="disabled"
    id="redo"
  >
    Redo
  </button>
   
  <h4>
    You don't have any Todos yet
  </h4>
   
  <ol />
</div>
`;
